pytensor/tensor/rewriting/math.py

```diff
@@ -1754,6 +1755,30 @@ def local_reduce_broadcastable(fgraph, node):
                 # -- in this case we can remove the reduction completely
                 return [new_reduced.astype(odtype)]
 
+@register_canonicalize
+@register_uncanonicalize
+@register_specialize
+@node_rewriter([Sum, Prod])
```

Comment:
When there's nothing special about Sum and Prod we should apply the rewrites to all CAReduce operations, which also include stuff like Max, All, Any, ...

See the related PR I linked to in the comment

Reply:
Oops, 2nd time. I did not see this one show up in #59 😄.

Reply:
No they are not the same exactly.

There's a rewrite for reduction along axis0 for a sum along axis 0.

My PR extends this to any axis.

There's then the question of multiple axis, of which axis=None is the most extreme (all axes). This PR can cover that case. 

We may also want to think about multiple but not all axis. In what cases can we reduce first and join later?